File: Controllers/BlockchainController.php
<?php
    //Controlador creado para manejar la vista 
    class BlockchainController{
        private $model;

        //Función para conectar con el modelo que utiliza la bd
        public function __construct(){
            require_once __DIR__ . '/../Models/Blockchain.php';
            $this->model = new Blockchain;
        }

        // Función para mostrar el usuario recien creado
        public function show($id)
        {
            return ($this->model->show($id) != false) ? $this->model->show($id) : header("Location:index.php");
        }


        //Esta función será para mostrar los detalles de cada usuario cuando se crea
        public function see()
        {
            return ($this->model->see()) ? $this->model->see() : false;
        }
    }
?>

File: Controller/blockchain_controller.py
#Controller para manejar el blockchain, en el cual se muestra el historial y los hashes
from Model.blockchain import Blockchain

class BlockchainController:
    def __init__(self):
        self.model = Blockchain()

    #Muestra los detalles del usuario cuando se crea
    def show(self, id):
        return self.model.show(id)
    
    #Muestra el historial del botón
    def see(self):
        return self.model.see()
    
    #Función para mostrar el historial de los hashes (hash anterior y hash actual)
    def get_block_by_hash(self, block_hash):
        conn = self.model.db.conexion() 
        cursor = conn.cursor(dictionary=True)

        sql = "SELECT * FROM blockchain WHERE hash = %s LIMIT 1"
        cursor.execute(sql, (block_hash,))
        block = cursor.fetchone()

        cursor.close()
        conn.close()

        return block